tests/arg-set.earth

```diff
@@ -0,0 +1,38 @@
+VERSION --arg-scope-and-set 0.7
+FROM alpine:3.15
+
+ARG --global foo = bar
+SET foo = ${foo} phone
+
+ARG foo = sports
+SET foo = ${foo} car
+
+RUN echo $foo
+
+test:
+    ARG bar = baz
+    SET bar = ${bar}inga
+    RUN echo $bar
```

Comment:
Since this deals a lot with how `ARG`s are assigned, a lot of the tests are overriding the `ARG` values - so I had to do a bit more work to get the tests to use `test`. That said, I think I figured it out. I've updated all of the happy paths to use `test` rather than `echo/output_contains`.